### ISSUE TITLE
test_algorithmid failure in master

### ISSUE THREAD

**Issue by mattcaswell** [openssl/openssl#14084]
The new algorithmid test is failing for me in master.

After a default build (`./config -d --strict-warnings`), I get this:

````
$ make TESTS=test_algorithmid test
make depend && make _tests
make[1]: Entering directory '/home/matt/dev/openssl-write'
make[1]: Leaving directory '/home/matt/dev/openssl-write'
make[1]: Entering directory '/home/matt/dev/openssl-write'
( SRCTOP=. \
  BLDTOP=. \
  PERL="/usr/bin/perl" \
  FIPSKEY="f4556650ac31d35461610bac4ed81b1a181b2d8a43ea2854cbae22ca74560813" \
  EXE_EXT= \
  /usr/bin/perl ./test/run_tests.pl test_algorithmid )
06-test_algorithmid.t .. 
    # ERROR: (memory) 'algid_legacy == algid_prov' failed @ test/algorithmid_test.c:187
    # --- algid_legacy
    # +++ algid_prov
    # 0000:-300d06092a864886 f70d01010b0500
    # 0000:+300b06092a864886 f70d01010b
    #         ^^                       
    # 
    # ERROR: (ptr) 'OSSL_PARAM_locate_const(gettable_params, ALGORITHMID_NAME) != NULL' failed @ test/algorithmid_test.c:79
    # 0x0
    # INFO:  @ test/algorithmid_test.c:80
    # The rsaEncryption provider keymgmt appears to lack support for algorithm-id.  Skipping...
    # ERROR: (ptr) 'OSSL_PARAM_locate_const(gettable_params, ALGORITHMID_NAME) != NULL' failed @ test/algorithmid_test.c:79
    # 0x0
    # INFO:  @ test/algorithmid_test.c:80
    # The rsaEncryption provider keymgmt appears to lack support for algorithm-id.  Skipping...
    not ok 1 - test_x509_files
# ------------------------------------------------------------------------------
../../util/wrap.pl ../../test/algorithmid_test -x509 ../../test/certs/ee-cert.pem ../../test/certs/ca-cert.pem => 1
not ok 1
06-test_algorithmid.t .. 1/? ---------------------------------------------------
    # ERROR: (memory) 'algid_legacy == algid_prov' failed @ test/algorithmid_test.c:187
    # --- algid_legacy
    # +++ algid_prov
    # 0000:-300d06092a864886 f70d01010b0500
    # 0000:+300b06092a864886 f70d01010b
    #         ^^                       
    # 
    # ERROR: (ptr) 'OSSL_PARAM_locate_const(gettable_params, ALGORITHMID_NAME) != NULL' failed @ test/algorithmid_test.c:79
    # 0x0
    # INFO:  @ test/algorithmid_test.c:80
    # The rsaEncryption provider keymgmt appears to lack support for algorithm-id.  Skipping...
    # ERROR: (ptr) 'OSSL_PARAM_locate_const(gettable_params, ALGORITHMID_NAME) != NULL' failed @ test/algorithmid_test.c:79
    # 0x0
    # INFO:  @ test/algorithmid_test.c:80
    # The rsaEncryption provider keymgmt appears to lack support for algorithm-id.  Skipping...
    not ok 1 - test_x509_files
# ------------------------------------------------------------------------------
../../util/wrap.pl ../../test/algorithmid_test -x509 ../../test/certs/ee-cert2.pem ../../test/certs/ca-cert2.pem => 1
not ok 5
# ------------------------------------------------------------------------------
    # ERROR: (memory) 'algid_legacy == algid_prov' failed @ test/algorithmid_test.c:187
    # --- algid_legacy
    # +++ algid_prov
    # 0000:-300d06092a864886 f70d01010b0500
    # 0000:+300b06092a864886 f70d01010b
    #         ^^                       
    # 
    # ERROR: (ptr) 'OSSL_PARAM_locate_const(gettable_params, ALGORITHMID_NAME) != NULL' failed @ test/algorithmid_test.c:79
    # 0x0
    # INFO:  @ test/algorithmid_test.c:80
    # The id-ecPublicKey provider keymgmt appears to lack support for algorithm-id.  Skipping...
    # ERROR: (ptr) 'OSSL_PARAM_locate_const(gettable_params, ALGORITHMID_NAME) != NULL' failed @ test/algorithmid_test.c:79
    # 0x0
    # INFO:  @ test/algorithmid_test.c:80
    # The rsaEncryption provider keymgmt appears to lack support for algorithm-id.  Skipping...
    not ok 1 - test_x509_files
# ------------------------------------------------------------------------------
../../util/wrap.pl ../../test/algorithmid_test -x509 ../../test/certs/server-ecdsa-brainpoolP256r1-cert.pem ../../test/certs/rootcert.pem => 1
not ok 6
# ------------------------------------------------------------------------------
06-test_algorithmid.t .. Dubious, test returned 3 (wstat 768, 0x300)
Failed 3/11 subtests 

Test Summary Report
-------------------
06-test_algorithmid.t (Wstat: 768 Tests: 11 Failed: 3)
  Failed tests:  1, 5-6
  Non-zero exit status: 3
Files=1, Tests=11,  1 wallclock secs ( 0.02 usr  0.01 sys +  0.38 cusr  0.06 csys =  0.47 CPU)
Result: FAIL
make[1]: *** [Makefile:3257: _tests] Error 1
make[1]: Leaving directory '/home/matt/dev/openssl-write'
make: *** [Makefile:3254: tests] Error 2
````

**Comment by levitte** [comment]
Yes, that will be fixed as soon as #14030 is in...  I'm merging it in now

**Comment by levitte** [comment]
There, #14030 is in.  Closing this, and please rebase your branches
